bot/cogs/snekbox.py

```diff
@@ -199,32 +203,64 @@ def get_status_emoji(results: dict) -> str:
 
         log.info(f"Received code from {ctx.author} for evaluation:\n{code}")
 
-        self.jobs[ctx.author.id] = datetime.datetime.now()
-        code = self.prepare_input(code)
+        while True:
+            self.jobs[ctx.author.id] = datetime.datetime.now()
+            code = self.prepare_input(code)
 
-        try:
-            async with ctx.typing():
-                results = await self.post_eval(code)
-                msg, error = self.get_results_message(results)
-
-                if error:
-                    output, paste_link = error, None
-                else:
-                    output, paste_link = await self.format_output(results["stdout"])
-
-                icon = self.get_status_emoji(results)
-                msg = f"{ctx.author.mention} {icon} {msg}.\n\n```py\n{output}\n```"
-                if paste_link:
-                    msg = f"{msg}\nFull output: {paste_link}"
-
-                response = await ctx.send(msg)
-                self.bot.loop.create_task(
-                    wait_for_deletion(response, user_ids=(ctx.author.id,), client=ctx.bot)
+            try:
+                async with ctx.typing():
+                    results = await self.post_eval(code)
+                    msg, error = self.get_results_message(results)
+
+                    if error:
+                        output, paste_link = error, None
+                    else:
+                        output, paste_link = await self.format_output(results["stdout"])
+
+                    icon = self.get_status_emoji(results)
+                    msg = f"{ctx.author.mention} {icon} {msg}.\n\n```py\n{output}\n```"
+                    if paste_link:
+                        msg = f"{msg}\nFull output: {paste_link}"
+
+                    response = await ctx.send(msg)
+                    self.bot.loop.create_task(
+                        wait_for_deletion(response, user_ids=(ctx.author.id,), client=ctx.bot)
+                    )
+
+                    log.info(f"{ctx.author}'s job had a return code of {results['returncode']}")
+            finally:
+                del self.jobs[ctx.author.id]
+
+            try:
+                _, new_message = await self.bot.wait_for(
+                    'message_edit',
+                    check=partial(predicate_eval_message_edit, ctx),
+                    timeout=10
                 )
+                await ctx.message.add_reaction('🔁')
+                await self.bot.wait_for(
+                    'reaction_add',
+                    check=partial(predicate_eval_emoji_reaction, ctx),
+                    timeout=10
+                )
+
+                log.info(f"Re-evaluating message {ctx.message.id}")
+                code = new_message.content.split(' ', maxsplit=1)[1]
+                await ctx.message.clear_reactions()
```

Comment:
This is always called, so you can move it outside of the loop.

Reply:
We want to call it after each eval, it needs to stay inside the loop

Reply:
Yes I agree, I meant that you call this inside `try` and also inside `except` so you can call this line outside of the `try except` altogether.

Reply:
Upon further discussion, there is a `return` in the `except` block, moving `await ctx.message.clear_reactions()` outside will make it not clear any reactions if the exception block is hit, so the code is good as is, and so I'll mark this as resolved.